dali/operators/generic/squeeze.cc

```diff
@@ -0,0 +1,106 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include <vector>
+
+#include "dali/core/math_util.h"
+#include "dali/core/static_switch.h"
+#include "dali/core/tensor_layout.h"
+#include "dali/core/tensor_shape_print.h"
+#include "dali/operators/generic/squeeze.h"
+#include "dali/pipeline/data/views.h"
+
+namespace dali {
+
+DALI_SCHEMA(Squeeze)
+  .DocStr(R"code(Removes the dimensions given as ``axes`` or ``axis_names``.
+
+It's an error to remove a dimension that would cause the total volume to change.)code")
+  .NumInput(1)
+  .NumOutput(1)
+  .InputDox(0, "data", "TensorList", "Data to be squeezed")
+  .PassThrough({{0, 0}})
+  .AllowSequences()
+  .SupportVolumetric()
+  .AddOptionalArg<int>("axes", R"code(Indices of dimensions which should be removed.
+
+All squeezed dimensions should have size 1, unless the total volume of the tensor is 0 before and after squeeze.
+All indices must be in the range of valid dimensions of the input)code", std::vector<int>(), true)
+  .AddOptionalArg("axis_names", R"code(Layout columns which should be removed.
+  
+All squeezed dimensions should have size 1, unless the total volume of the tensor is 0 before and after squeeze.
+All layout names should be present in data layout.)code", TensorLayout(""));
+
+template <typename Backend>
+Squeeze<Backend>::Squeeze(const OpSpec &spec)
+    : Reshape<Backend>(spec, typename Reshape<Backend>::BypassInit()) {
+  axes_ = spec.GetRepeatedArgument<int>("axes");
+  axis_names_ = spec.GetArgument<TensorLayout>("axis_names");
+
+  DALI_ENFORCE(spec.HasArgument("axes") + spec.HasArgument("axis_names") == 1,
+    make_string("Provided both axes and axis_names argument"));
+
+  this->use_src_dims_ = true;
+}
+
+template <typename Backend>
+bool Squeeze<Backend>::SetupImpl(std::vector<OutputDesc> &output_desc, const Workspace &ws) {
+  output_desc.resize(1);
+  this->SetOutputType(ws);
+
+  GenerateSrcDims(ws);
+  Reshape<Backend>::CalculateOutputShape(ws);
+
+  output_desc[0].type = *(this->output_type_);
+  output_desc[0].shape = this->output_shape_;
+  // return false, because we don't want the executor to allocate anything
+  // - this operator returns pointer to input memory
+  return false;
+}
+
+template <typename Backend>
+void Squeeze<Backend>::GenerateSrcDims(const Workspace &ws) {
+  auto &in = ws.template InputRef<Backend>(0);
+  const auto &input_shape = in.shape();
+  const int ndim = input_shape.sample_dim();
+  auto in_layout = in.GetLayout();
+  DALI_ENFORCE(in_layout.size() == ndim || in_layout.empty(),
+      make_string("Layout for data has ",
+      in_layout.size(), " elements but data has ", ndim, " dimensions."));
+
+  this->src_dims_.clear();
+  auto axes = axis_names_.empty() ? axes_ : GetDimIndices(in_layout, axis_names_).to_vector();
+  std::sort(axes.begin(), axes.end());
+  axes.erase(std::unique(axes.begin(), axes.end()), axes.end());
```

Comment:
I think that specifying an axis multiple times should be an error, not something swept under the carpet.